cli/internal/repository/ocm/compatibility.go

```diff
@@ -136,3 +147,61 @@ func NewComponentVersionRepositoryForComponentProvider(ctx context.Context,
 	}
 	return nil, nil
 }
+
+// NewComponentVersionRepositoryForComponentProvider creates a new ComponentVersionRepositoryForComponentProvider based on the provided
+// component reference and configuration.
+// If a compref.Ref is provided, it will be used to create a compRefProvider.
+// If a genericv1.Config is provided, it will be used to create either a fallback resolver provider (deprecated)
+// or a path matcher resolver provider.
+// If both types are configured, an error will be returned.
+// If neither a componentReference nor a configuration is provided, an error will be returned.
+// As a fallback, this constructor adds the compref as a fallback entry as both
+// resolverruntime.Resolver (lowest priority) and resolverspec.Resolver (highest priority) depending on the configuration type.
+func NewComponentVersionRepositoryForComponentProvider(ctx context.Context,
```

Comment:
what happens if both a ref and a config are defined? The documentation makes this a bit unclear